### DISCUSSION TITLE
Pinning checkbox column

### DISCUSSION THREAD

**Discussion by sotatek-HarryPhi** [KevinVandy/material-react-table#207] [Q&A] [answered]
I'm finding out how to pin (freeze) the checkbox column. In our package, checkbox column is pinned by using the column id but there's no documents relating to the id of checkbox column, please help us if you have some solutions. Many thanks!!!

**Comment by sotatek-HarryPhi** [comment]
@KevinVandy Please help me if you have some solutions. Thank you so much.

**Reply by KevinVandy** [reply]
There's docs for exactly how to do that here https://www.material-react-table.com/docs/guides/display-columns

No need to spam a bunch of reactions on posts to get attention.

**Reply by sotatek-HarryPhi** [reply]
Thank you @KevinVandy

**Reply by sotatek-HarryPhi** [reply]
It works properly, thank you so much!

**Reply by vipindigiqt** [reply]
Hi @sotatek-HarryPhi 

Can you please share the exact code you did to pin the checkbox column (1st column)?

Thanks!

**Comment by KevinVandy** [comment, marked as answer]
https://www.material-react-table.com/docs/guides/display-columns

**Reply by vipindigiqt** [reply]
> https://www.material-react-table.com/docs/guides/display-columns

Hi @KevinVandy, sorry to bother you!

Actually, I checked out the docs you linked here, but can't figure out how to make my Row selection column pinned (want to pin it to the first column)

Can you please guide me with a gist here?


Thanks!

**Reply by KevinVandy** [reply]
```jsx
<MaterialReactTable
  displayColumnDefOptions={{
    'mrt-row-select': {
      enablePinning: true //enable select column to be pinned in the ui
    }
  }}
  inintialState={{
    columnPinning: { left: { 'mrt-row-select': true } } //pin select column by default
  }}
/>
```
